lib/std/sort.zig

```diff
@@ -89,23 +93,30 @@ pub fn heapContext(a: usize, b: usize, context: anytype) void {
     }
 }
 
-fn siftDown(a: usize, root: usize, n: usize, context: anytype) void {
-    var node = root;
-    while (true) {
-        var child = a + 2 * (node - a) + 1;
-        if (child >= n) break;
+fn siftDown(a: usize, cur: usize, b: usize, context: anytype) void {
+    var child = (math.mul(usize, cur - a, 2) catch return) + a + 1;
+    // When we don't overflow from the multiply, the above expression equals (2*cur) - (2*a) + a + 1
+    // The `+ a + 1` is safe because:
+    //  for `a > 0` then `2a >= a + 1`.
+    //  for `a = 0`, the expression equals `2*cur+1`. `2*cur` is an even number, therefore adding 1 is safe.
 
-        // choose the greater child.
-        child += @intFromBool(child + 1 < n and context.lessThan(child, child + 1));
+    // stop if we overshot the boundary
+    if (child < b) {} else return;
```

Comment:
Nitpick: this looks weird to me, I think the following would be more idiomatic if you want to keep the comparison operator `<` instead of ` >=`
```suggestion
    if (!(child < b)) return;
```

Reply:
Why not `if (child >= b) return;`?

Reply:
I originally had `if (child >= b) return`, but I personally felt like it was easier to think about the overflow conditions when it is flipped the other way around.